### DISCUSSION TITLE
Belgium, flanders

### DISCUSSION THREAD

**Discussion by MrOzzOnline** [JaccoR/hass-entso-e#37] [Price Modifyer Templates]
New prices starting from 1/1/2023

The price for injection (= injectietarief) is without tax and costs. You can use the unmodified value.

{% set s = { 
	"bijz_accijns": 0.014416, 
	"energiebijdrage": 0.0020417,
	"groenestroom_wkk": 0.02184,
	"kwh_tarief":  0.0374193,
	"VAT": 1.06, 
	"marge": 0.002040 
	} %} 
{{((current_price + s.marge) * s.VAT + s.bijz_accijns + s.energiebijdrage + s.groenestroom_wkk + s.kwh_tarief) | float}}

Remarks:
* The value for "kwh_tarief" is different by region. These prices are for the region Antwerp.
* "marge" is the additional margin the supplier takes on the epex spot prices.
* These prices don't include the average peak costs.

Costs per region:
![image](https://user-images.githubusercontent.com/115424166/210185629-ef9d97c9-ebdc-4094-8410-a9145cea88a8.png)

You can find these values on the cost sheets of your energy supplier (for example: https://www.engie.be/nl/energie/elektriciteit-gas/prijzen-voorwaarden/)

The distribution cost details can be found here:
https://www.vreg.be/nl/periodieke-nettarieven-elektriciteit-en-aardgas-2023

**Comment by elRadix** [comment]
Thank you!

**Comment by TsaneTips** [comment]
I think the cost voor "bijdrage groene stroom & WKK" is missing?

{% set s = {
"distributiekosten": 0.0787,
"transportkosten": 0.0115,
"bijz_accijns": 0.014416,
"energiebijdrage": 0.002042,
"groenestroom_wkk": 0.002544,
"VAT": 1.06
}
%}
{{(current_price * s.VAT + s.distributiekosten + s.transportkosten + s.bijz_accijns + s.energiebijdrage + s.groenestroom_wkk) | float}}

**Reply by MrOzzOnline** [reply]
Thanks, I missed it. I'll edit the original post.

**Reply by Stinosko** [reply]
groenestroom_wkk should be 0.02544 or am i wrong?

**Reply by MrOzzOnline** [reply]
> groenestroom_wkk should be 0.02544 or am i wrong?

Nope, you are right. I'll get it 100% right sooner or later ..

**Reply by blowk** [reply]
From 2023, the distribution cost will be calculated different.
Then it wil be changed to the capacity tariff, but for now I don't have the exact calculation for it.

**Reply by elRadix** [reply]
@MrOzzOnline you forgot to add a comma at the vat line

**Reply by MrOzzOnline** [reply]
> @MrOzzOnline you forgot to add a comma at the vat line

thanks, corrected it

**Comment by elRadix** [comment]
it's a bit strange that the actual price in vtest.vreg.be is about 70c/kWh and in Entso-e around 21c/kWh... anything missing? 


![IMG_20221023_214954_090](https://user-images.githubusercontent.com/6258894/197414805-b77e605a-af69-4d9c-b6d1-d825b43a762f.jpg)
![SmartSelect_20221023_214810_Chrome](https://user-images.githubusercontent.com/6258894/197414806-0c86ccde-34c8-45fe-ab61-683faebb272b.jpg)

**Reply by Stinosko** [reply]
On the website of Engie you can consult the base price for each hour tommorow, those match with the prices on Entso: 
https://www.engie.be/nl/energie/elektriciteit-gas/prijzen-voorwaarden
"Onze commerciële tarieven" => "dagprijzen" in the right kolom.

**Reply by MrOzzOnline** [reply]
> it's a bit strange that the actual price in vtest.vreg.be is about 70c/kWh and in Entso-e around 21c/kWh... anything missing?

No. The vreg price takes into account the currently higher priced contracts which have a fixed price for 1 or 3 months.

**Reply by pyto00** [reply]
1. U are comparing 2 totally different pricing schema`s.
 Luminius =  Variable - Changes every 1 or 3 months
 Enstoe graph = Dynamic - Changes every hour

2. c€/kWh =/= €/kWh

**Comment by Stinosko** [comment]
Does one of you know what price you actually get when you put electricity back on the net with a dynamic contract, i assume it would be something like this?

current_price - s.distributiekosten - s.transportkosten

**Reply by MrOzzOnline** [reply]
> Does one of you know what price you actually get when you put electricity back on the net with a dynamic contract, i assume it would be something like this?
> 
> current_price - s.distributiekosten - s.transportkosten

It's the plain espot price. See remark 3 in the dynamic contract conditions:

"De prijs voor de geïnjecteerde elektriciteit wordt geïndexeerd op basis
van de parameter eSpot, zoals gedefinieerd in (2). De formules excl.
btw van toepassing voor Oktober 2022 zijn:
- Enkelvoudig = 0,0000 + 0,1000 x eSpot"

**Reply by elRadix** [reply]
then what would be nice to have another sensor for "Injectie prijs"

**Reply by MrOzzOnline** [reply]
> then what would be nice to have another sensor for "Injectie prijs"

It's already possible using templates, but if you want to use the attributes too (for a graph for example) the template could become quite complex. I'm working on one and I'll post it when I have it ready.

As an alternative quick fix you can pull a second set of data using https://github.com/custom-components/nordpool
It isn't possible to include a custom VAT in this integration, so it's a perfect fit for the injection prices.

At the moment I have this combining both integrations:
![image](https://user-images.githubusercontent.com/115424166/197467429-f1159729-bbb4-4103-9f91-ee7bb0602127.png)

**Reply by MrOzzOnline** [reply]
Code for the graph above:

```
type: custom:apexcharts-card
graph_span: 48h
span:
  start: day
all_series_config:
  stroke_width: 1
  show:
    in_brush: true
now:
  show: true
  label: Nu
header:
  show: true
  title: Electriciteitsprijzen - afname (€/kwh)
experimental:
  brush: true
series:
  - entity: sensor.current_electricity_market_price
    name: Prijs nu
    stroke_width: 2
    float_precision: 4
    type: column
    opacity: 1
    color: ''
    data_generator: |
      return entity.attributes.prices_today.map((record, index) => {
        return [record.time, record.price];
      });
  - entity: sensor.current_electricity_market_price
    name: Zelfde tijd morgen
    stroke_width: 2
    float_precision: 4
    type: column
    opacity: 0.7
    color: ''
    data_generator: |
      return entity.attributes.prices_tomorrow.map((record, index) => {
        return [record.time, record.price];
      });
  - entity: sensor.nordpool_kwh_be_eur_5_10_0
    name: Injectie nu
    opacity: 1
    stroke_width: 2
    float_precision: 4
    type: column
    color: ''
    data_generator: |
      return entity.attributes.raw_today.map((start, index) => {
        return [new Date(start["start"]).getTime(), entity.attributes.raw_today[index]["value"]];
      });
  - entity: sensor.nordpool_kwh_be_eur_5_10_0
    name: Injectie morgen
    stroke_width: 2
    float_precision: 4
    type: column
    color: ''
    opacity: 0.5
    data_generator: |
      return entity.attributes.raw_tomorrow.map((start, index) => {
        return [new Date(start["start"]).getTime(), entity.attributes.raw_tomorrow[index]["value"]];
      });
apex_config:
  yaxis:
    opposite: false
    reversed: false
    logarithmic: false
    decimalsInFloat: 4
    labels:
      show: true
    tooltip:
      enabled: true
    crosshairs:
      show: true
  xaxis:
    labels:
      show: true
      rotate: -45
    logarithmic: true
```

**Reply by MrOzzOnline** [reply]
> > then what would be nice to have another sensor for "Injectie prijs"
> 
> It's already possible using templates, but if you want to use the attributes too (for a graph for example) the template could become quite complex. I'm working on one and I'll post it when I have it ready.

Template code (put it in configuration.yaml) to extract the additional costs (VAT + distribution) using both this sensor and nordpool:

```
template:
    - sensor:
        - name: "distributiekosten_en_btw"
          unit_of_measurement: "€/kWh"
          state: >
            {% set totaal_prijs = states('sensor.current_electricity_market_price') | float %}
            {% set netto_prijs = states('sensor.nordpool_kwh_be_eur_5_10_0') | float %}
          
            {{ totaal_prijs - netto_prijs | float }}
```

**Reply by JaccoR** [reply]
I am currently working on a way to add multiple instances of this integration. This way you could add two instances, one with a template for feed back prices and one with the normal prices!

**Comment by Stinosko** [comment]
I calculate the actually price at the chart.

```type: custom:apexcharts-card
graph_span: 24h

span:
  start: day
now:
  show: true
  label: Now
header:
  show: true
  title: Electriciteitsprijzen Vandaag (€/kwh)
series:
  - entity: sensor.current_electricity_market_price
    name: "Afname prijs"
    stroke_width: 2
    float_precision: 3
    type: column
    opacity: 1
    color: ''
    data_generator: >
      return entity.attributes.prices_today.map((record, index) => {
      const s = {"distributiekosten": 0.0942,
        "transportkosten": 0.0126,
        "bijz_accijns": 0.014416,
        "energiebijdrage": 0.002042,
        "groenestroom_wkk": 0.02544,
        "VAT": 1.06}
      return [record.time, (record.price * s.VAT + s.distributiekosten +
      s.transportkosten + s.bijz_accijns + s.energiebijdrage +
      s.groenestroom_wkk)];

      });
  - entity: sensor.current_electricity_market_price
    name: "Teruggave prijs"
    stroke_width: 2
    float_precision: 3
    type: column
    opacity: 1
    color: ''
    data_generator: >
      return entity.attributes.prices_today.map((record, index) => {
      return [record.time, record.price];
      });

yaxis:
  - min: 0
    decimals: 3
    align_to: 0.1
```
![afbeelding](https://user-images.githubusercontent.com/43827063/197601349-896a4245-9690-4fb6-84e1-95a53d75cff5.png)

**Comment by JaccoR** [comment]
In the new release, VAT can be added as an config option. This means the template does not need the VAT anymore.

**Comment by elRadix** [comment]
@MrOzzOnline can you please adjust the template since the new release and maybe add one for 'Injectie prijs'

**Reply by MrOzzOnline** [reply]
> @MrOzzOnline can you please adjust the template since the new release and maybe add one for 'Injectie prijs'

I noticed the calculation of VAT is not correct. See https://github.com/JaccoR/hass-entso-e/issues/54.

'injectie prijs' is the unmodified value and doesn't need a template.

**Comment by MrOzzOnline** [comment]
I noticed the margin the supplier takes (only Engie at the moment) isn't included in the price template. According to their conditions:

![image](https://user-images.githubusercontent.com/115424166/199012340-577e33c6-217d-42dd-8d4c-87e3d12e6ba4.png)

The 0,2040 is actually c€/kWh while the spot price is in €/MWh which is why they are multiplying it by 0,1 which is the result of /1000 (MWh to kWh) x 100 (€ to c€).

This integration gets the price in €/kWh so the actual margin used is 0,002040 €/kWh. This is before tax so the general formula becomes: (spot price x margin) x VAT + costs.

I changed the template in the first post.

**Reply by Stinosko** [reply]
They couldn't make any simpler 🙄

It guess it will become more difficult when they start calculating the distribution cost by the peak consumption...

**Comment by MrOzzOnline** [comment]
The way the additional costs are calculated has changed. New template in the first post.

**Reply by Stinosko** [reply]
How do you know that the VATs are added after all other cost are calculated?
The pdf of engie explaining the costs states `prijzen inclusief 5% BTW` so i assumed that most cost are already calculated with the 6% VATS?
So i was calculating the cost by: 
```
(current_price + s.marge) * s.VAT + s.federale_accijns + s.energiebijdrage + s.groenestroom_wkk + s.kwh_tarief 
```

**Reply by BliXem1** [reply]
Correct, but it's 6% BTW and not 5%. So most prices are included with VAT but the current_price it's not.
The latest document of Engie Dynamic has been uploaded: https://www.engie.be/api/engie/be/ms_1/pricing/v1/public/pricesAndConditionsPDF?document=E_DYNAMIC_R_GREY_C_I_12_V_N&monthOffset=0&segment=R&language=N

**Reply by Stinosko** [reply]
Oooh yes 6% BTW, my mistake

**Reply by BliXem1** [reply]
> Oooh yes 6% BTW, my mistake

I've still got this:

{{ ((current_price + marge) * VAT + distributiekosten + transportkosten + bijz_accijns + energiebijdrage + groenestroom_wkk) |float }}

The current_price + marge * VAT and then + everything. But dunno why @MrOzzOnline did change it in his post. I know about the changes of the peak price.

**Comment by BliXem1** [comment]
OK I changed it now too:

```
      {% set VAT = 1.06 %}
      {% set kwh_tarief = 0.0394475 %}
      {% set bijz_accijns = 0.0144160 %}
      {% set energiebijdrage = 0.0020417 %}
      {% set groenestroom_wkk = 0.02714 %}
      {% set marge = 0.002040 %}
      {{ ((current_price + marge) * VAT + kwh_tarief + bijz_accijns + energiebijdrage + groenestroom_wkk) |float }}
```

I'm using nordpool for now because I don't have access to the API yet. See screenshot, tomorrow will be a expansive day?
https://ibb.co/fqvWnG4

**Comment by MrOzzOnline** [comment]
Allright, I did some more digging:

You can find the full details of the distribution fees here:
https://www.vreg.be/nl/periodieke-nettarieven-elektriciteit-en-aardgas-2023

"Totaal kWh-tarief" mentioned at the Engie price sheet for distribution is actually:
(kWh-tarief + openbaredienstverplichtingen (normaal of nacht) + Tarieven voor de toeslagen + overige transmissienetkosten) * VAT
For Antwerp: (0,0090800 + 0,0215095 + 0,0011539 + 0,0035578) * 1.06 = 0,037419272 €/kWh or 3.74193 c€/kWh as in the sheet.

I'll update the first post.

**Comment by BliXem1** [comment]
I have nordpool and the entso api running, but there is a big difference in price, any idea why? And yes, both have the right calculation (kwh-tarief, etc etc). Which one is the right oned to check for the Belgium energy prices?

See screenshot: https://ibb.co/L09Mqzz

**Reply by MrOzzOnline** [reply]
> I have nordpool and the entso api running, but there is a big difference in price, any idea why? And yes, both have the right calculation (kwh-tarief, etc etc). Which one is the right oned to check for the Belgium energy prices?
> 
> See screenshot: https://ibb.co/L09Mqzz

If you did everything correct the prices should be exactly the same. Looking at the values, your nordpool graph seems to be wrong. the prices are to high. Tomorrow at 0:00 the price should be around €0,22 including VAT, €0,13 without extra costs.

**Reply by BliXem1** [reply]
Thanks for your reply @MrOzzOnline, maybe you see here the problem?
```

platform: nordpool
VAT: False
currency: "EUR"
price_in_cents: true
low_price_cutoff: 0.9
region: "BE"
precision: 3
price_type: kWh
additional_costs: >
{% set VAT = 1.06 %}
{% set kwh_tarief = 0.0394475 %}
{% set bijz_accijns = 0.0144160 %}
{% set energiebijdrage = 0.0020417 %}
{% set groenestroom_wkk = 0.02714 %}
{% set marge = 0.002040 %}
{{ ((current_price + marge) * VAT + kwh_tarief + bijz_accijns + energiebijdrage + groenestroom_wkk) |float }}
```

I'm gonna check this out what is wrong :).

**Reply by MrOzzOnline** [reply]
the additional costs value is in  euro while you values are in cents?

**Reply by BliXem1** [reply]
Well.. It's frustrating me now. I got the same results as Nordpool with the additional costs value, added in entso-e:

```
type: custom:apexcharts-card
graph_span: 24h
span:
  start: day
  offset: +1d
now:
  show: true
  label: Now
header:
  show: true
  title: Electriciteitsprijzen Morgen (€/kwh)
series:
  - entity: sensor.average_electricity_price_today
    stroke_width: 2
    float_precision: 3
    type: column
    opacity: 1
    color: ''
    data_generator: >
      return entity.attributes.prices_today.map((record, index) => { const s =
      {"kwh_tarief": 0.0394475,
        "bijz_accijns": 0.0144160,
        "energiebijdrage": 0.0020417,
        "groenestroom_wkk": 0.02714,
        "marge": 0.002040,
        "VAT": 1.06}
      return [record.time, ((record.price + s.marge) * s.VAT + s.kwh_tarief +
      s.bijz_accijns + s.energiebijdrage + s.groenestroom_wkk)];

      });
yaxis:
  - min: 0
    decimals: 3
    align_to: 0.1


```
See the results here: https://ibb.co/GkVsdQy
These are the results for tomorrow.......  3 Jan with nordpool 27,8 and with ento-se 28,1. Maybe you can give me some examples how you setup this and I can adjust it then?

**Reply by MrOzzOnline** [reply]
Both prices are wrong. The price for tomorrow at 23:00 is €0.177 (in Antwerp).

You need to remove all variables and check what is wrong. Do you get the correct value without the template? Something wrong with your chart config? 

To me it looks like you've added the template twice. Once in the config, once in the chart, which would explain the error.

**Reply by BliXem1** [reply]
Yeah, removed the extension and added it again, without any VAT or add. costs, it's not showing a correct price within ents-o, within the nordpool it does it correct and it tells me what you did say, tomorrow 0.00 13cents. So Nordpool is correct now.
When I check ents-o it's showing 0,223 cents at 0.00 tomorrow. I really don't understand it anymore.

```
type: custom:apexcharts-card
graph_span: 24h
span:
  start: day
  offset: +1d
now:
  show: true
  label: Now
header:
  show: true
  title: Electriciteitsprijzen Vandaag (€/kwh)
series:
  - entity: sensor.average_electricity_price_today
    name: Afname prijs
    stroke_width: 2
    float_precision: 3
    type: column
    opacity: 1
    color: ''
    data_generator: >
      return entity.attributes.prices_today.map((record, index) => { const s =
      {"distributiekosten": 0.0,
        "transportkosten": 0.0,
        "bijz_accijns": 0.0,
        "energiebijdrage": 0.0,
        "groenestroom_wkk": 0.0,
        "VAT": 0.00}
      return [record.time, (record.price + s.distributiekosten +
      s.transportkosten + s.bijz_accijns + s.energiebijdrage +
      s.groenestroom_wkk)];

      });
yaxis:
  - min: 0
    decimals: 3
    align_to: 0.1
```


There is no config within price modifyer and VAT is on 0.00 within the config.

I will check it tomorrow again...

**Reply by BliXem1** [reply]
Got it right now, there is a difference of 0.002 euro. Are the prices correct at your side @MrOzzOnline ?

I added this:

{% set correctie = 0.002 %}

Which fixes everything now :)

**Comment by Piloteer1** [comment]
How do you use new 'capaciteitstarief' in this template? As far as I know, it's replacing the distribution and transport kost.

**Reply by MrOzzOnline** [reply]
> How do you use new 'capaciteitstarief' in this template? As far as I know, it's replacing the distribution and transport kost.

It's not completely replacing it, just part of it. This rate is based on your monthly peak use and not on the current electricity usage. You can read the peak values using the https://www.home-assistant.io/integrations/dsmr_reader integration.
You can find the rates here: https://www.vreg.be/nl/periodieke-nettarieven-elektriciteit-en-aardgas-2023

This is not an Entso-e issue

**Comment by blowk** [comment]
> How do you use new 'capaciteitstarief' in this template? As far as I know, it's replacing the distribution and transport kost.

This is difficult because this a variable price according to your own energy use.
This extension is mainly for the price of the energy component and used with a dynamic energy contract.

**Reply by Piloteer1** [reply]
I know it's not an Entso-e issue as well as the extension is for Dynamic pricing. I was more looking into adding the minimum cost for 'capaciteitstarief' which is based upon 2,5 kWh. All usage above as peak will generate an small additional cost (in the assumption that you have your peak usage under control). So I'm more then happy if I could use the template to have at least the minimum cost. But I'm not sure which value to add in the template.  According to my contract, the capacity cost will be 43,6742 €/kW/year. I assume this will be for the minimum of 2,5 kWh. But I have no idea which value to add in the template.

**Reply by blowk** [reply]
Is this the price without vat?
I think you could add 0,00498563927 euro as an additional cost. This is the price per kW per hour based on 2.5kW capacity.

**Reply by Piloteer1** [reply]
The price 43,6742 €/kW/year is including VAT. What is the calculation behind your output?
This seems to be a rather low cost. I assume, the price should be more or less match the current distribution and transport as in the original template. This is an assumption. I assume, the new cost will not be less than the previous ones for the standard 'capaciteitstarief' of 2,5 kWh. 
Unless the costs in the initial template are not per mWh?
"distributiekosten": 0.0787,
"transportkosten": 0.0115,

**Reply by blowk** [reply]
Sorry, my calculation is wrong. I thought the price was per kWh/year, but it's kW/year.
The cost will be the average of 12 months.
For example if the average capacity is 2.5, then the price is 2.5 * 43.6742.
2.5 is minimum. 
In order to add the cost, you need a HA helper with the average peak value (sum of all peak values from each month divided by 12). Multiply this with the cost (43.6742) and dividing this with 8.760 (365 days * 24 hours).
0,0124640982 euro per hour. This value will be dynamic based on the average peak.

**Reply by Piloteer1** [reply]
Thanks for the reply and thinking with me. I however would like to have the minimum price in the template.
I only use HA for standard integrations and very specific integrations. All other intelligence is all done in Node-red.
Maybe below assumption and calculation can be challenged?
In the my energy app (Engie), it is mentioned the 'peak without additional cost' = 0,63 kWh (notice, no longer kW but kWh).
May I assume, 0,63 kWh has a cost of € 109,1855 (= 43,6742 €/kW/year *2,5).
109,1855 / 0,63 = 173,3103
173,3103 / 365 / 24 = 0,019784283 / kWh to add in the template.
Any idea if this can be correct?

**Reply by blowk** [reply]
I think you can't use your 0,63 peak value. The minimum will still be 2.5 (although you have lower peak value), so you don't have to divided by 0.63.

**Reply by Piloteer1** [reply]
In the app with overview of price and usage, as explanation for 'capaciteitstarief' you can find the below paragraph (screenshot in Dutch).
2,5 kW Peak is equal to 0,63 kWh.
If your peak is below, you pay the standard fee of 43,6742 €/kW/year.
At least that is my understanding.

![image](https://user-images.githubusercontent.com/98615320/221952138-962cca17-7e7d-4d17-8458-e6ed27e8f494.png)

![image](https://user-images.githubusercontent.com/98615320/221952835-b74fe37a-7899-4333-80c9-68b2577d1ef9.png)

**Reply by blowk** [reply]
0.63kWh is per 15 minutes.
Dynamic price is per hour so the price has only be divided by 8760.

**Comment by weemaba999** [comment]
Does anyone know, using the ENTSO-E integration, why I have for the tomorrow prices : N/A instead of the prices in the chart ?

Thanks in advance !

Bart

**Reply by blowk** [reply]
Can you give us more information? Is it on the graph that you see N/A?

**Reply by weemaba999** [reply]
Hi @blowk,

It is indeed on the graph. Please  see below the code I am using:

`              - type: 'custom:apexcharts-card'
                graph_span: 24h
                span:
                  start: day
                all_series_config:
                  stroke_width: 1
                  show:
                    in_brush: true
                now:
                  show: true
                  label: Nu
                header:
                  show: true
                  title: Electriciteitsprijzen - afname (€/kwh)
                  show_states: true
                  standard_format: true
                  colorize_states: true
                chart_type: line
                series:
                  - entity: sensor.bart_weemaels_average_electricity_price_today
                    name: Prijs nu
                    float_precision: 2
                    type: column
                    opacity: 1
                    data_generator: |
                      return entity.attributes.prices_today.map((record, index) => {
                        return [new Date(record.time).getTime(), record.price];
                      });
                    color: '#4169E1'
                    show:
                      legend_value: false
                      in_header: before_now
                  - entity: sensor.bart_weemaels_average_electricity_price_today
                    name: Zelfde tijd morgen
                    float_precision: 2
                    type: column
                    opacity: 0.8
                    color: '#00BFFF'
                    data_generator: |
                      return entity.attributes.prices_tomorrow.map((record, index) => {
                        return [new Date(record.time).getTime()-86400000, record.price];
                      });
                    show:
                      legend_value: false
                      in_header: before_now

                apex_config:
                  datalabels:
                    enabled: true
                  yaxis:
                    opposite: false
                    reversed: false
                    logarithmic: false
                    decimalsInFloat: 3
                    min: 0
                    labels:
                      show: true
                    tooltip:
                      enabled: true
                    crosshairs:
                      show: true
                  xaxis:
                    labels:
                      show: true
                      rotate: -45`

**Reply by blowk** [reply]
If just want to display 48h time window (prices today and tomorrow), you can use this code below.

```
type: custom:apexcharts-card
graph_span: 48h
span:
  start: day
now:
  show: true
  label: Now
header:
  show: true
  title: 48h prices (€/kwh)
experimental:
  color_threshold: true
series:
  - entity: sensor.average_electricity_price_today
    stroke_width: 10
    float_precision: 3
    type: column
    opacity: 0.9
    color: '#1007f0'
    color_threshold:
      - value: 0.05
        color: lightgreen
      - value: 0.1
        color: green
      - value: 0.15
        color: orange
      - value: 0.2
        color: darkorange
      - value: 0.25
        color: black
      - value: 0.3
        color: purple
      - value: 0.35
        color: maroon
      - value: 0.4
        color: red
    show:
      legend_value: false
    data_generator: |
      return entity.attributes.prices.map((record, index) => {
        return [record.time, record.price];
      });
```

**Reply by weemaba999** [reply]
Hi @blowk,

That is indeed no problem, but still can 't see my second 'entity' / my second graphline / / the value for tomorrow same time.
What did you choose in the integration ? Sliding/ publish / default

Maybe there I have something wrong ?

KR,

Bart

**Reply by blowk** [reply]
I use publish. Never tried to other options.

**Reply by weemaba999** [reply]
Very weird, here too

I posted a screenshot on the ENTSO-E community page on HAS community:

https://community.home-assistant.io/t/custom-component-entso-e-day-ahead-energy-prices/467127/91

**Reply by weemaba999** [reply]
What I've also noticed, that in development tools > status , my entso-e sensors give pricing for today 2023-09-29 and for tomorrow 2023-09-28 ????!!!! how is that possible :-)

**Reply by blowk** [reply]
Which version do you run?

Have you tried to remove the integration and re-add it again?

**Reply by blowk** [reply]
I'm using v0.3.0

My settings are
![IMG_20230929_203141.jpg](https://github.com/JaccoR/hass-entso-e/assets/2361395/9d8acdba-bf55-4133-b98b-632662b21a85)

**Comment by ward0** [comment]
<img width="620" height="806" alt="image" src="https://github.com/user-attachments/assets/aea9afa3-2f6b-4b72-8fdd-f106902ac52c" />
calculation_mode : publish

Echter zie ik geen sensors, ook niet na reload:
<img width="1154" height="293" alt="image" src="https://github.com/user-attachments/assets/b9630ca8-c3d1-434c-aec1-44bc0148940f" />

Wat kan ik nog doen?
De enige error die ik zag:
entsoe: Error on device update!
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 807, in _async_add_entity
    await entity.async_device_update(warning=False)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1314, in async_device_update
    await self.async_update()
  File "/config/custom_components/entsoe/sensor.py", line 225, in async_update
    bucket_time(utcnow(), self.coordinator.period_minutes)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'EntsoeCoordinator' object has no attribute 'period_minutes'

**Reply by ward0** [reply]
Suddenly it started to work. Until the integration crashed my ha yesterday 😅

**Reply by blowk** [reply]
> It will be block in HA 2025.12.1 so it doesn't crash your HA. The integration needs to be fixed first and there is a PR ready for it.
